ckan/config/middleware/flask_app.py

```diff
@@ -341,3 +341,10 @@ def is_blueprint(mm):
         for blueprint in inspect.getmembers(module, is_blueprint):
             app.register_blueprint(blueprint[1])
             log.debug(u'Registered core blueprint: {0!r}'.format(blueprint[0]))
+            # print app.url_map
+            # building the routes and saving to config
+            # need to be improved
+            for rule in app.url_map.iter_rules():
+                config['routes.named_routes'].update({
+                    rule.endpoint: rule.rule
+                })
```

Comment:
How about:
```python
            for rule in app.url_map.iter_rules():
                if '.' not in rule.endpoint:
                    continue
                controller, action = rule.endpoint.split('.')
                route = {
                    rule.endpoint: {
                        'action': action,
                        'controller': controller,
                        'highlight_actions': action,
                        'needed': list(rule.arguments)
                    },

                }
                config['routes.named_routes'].update(route)
```
this will convert our blueprints to items quite similar to pylons routes